pkg/controller/perconaservermongodb/psmdb_controller.go

```diff
@@ -262,14 +263,12 @@ func (r *ReconcilePerconaServerMongoDB) Reconcile(request reconcile.Request) (re
 		}
 	}
 
-	for i, replset := range cr.Spec.Replsets {
-		// multiple replica sets is not supported until sharding is
-		// added to the operator
-		if i > 0 {
-			reqLogger.Error(nil, "multiple replica sets is not yet supported, skipping replset %s", replset.Name)
-			continue
-		}
+	repls := cr.Spec.Replsets
+	if cr.Spec.Sharding.Enabled && cr.Spec.Sharding.ConfigsvrReplSet != nil {
+		repls = append(repls, cr.Spec.Sharding.ConfigsvrReplSet)
+	}
 
+	for _, replset := range repls {
 		matchLabels := map[string]string{
 			"app.kubernetes.io/name":       "percona-server-mongodb",
 			"app.kubernetes.io/instance":   cr.Name,
```

Comment:
please add check for `"app.kubernetes.io/component": "mongod",`

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/439/commits/ee3a7fe9b56a9762d16bb730040e7ac96e677329